app/controllers/challenges/dashboard/allowed_groups_controller.rb

```diff
@@ -0,0 +1,51 @@
+module Challenges
```

Comment:
When required groups will be stored inside `Challenge` field, dedicated views will not be needed - only an additional field in the challenge form.